parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh

```diff
@@ -257,6 +261,29 @@ ensureRunc() {
 
 cleanUpGPUDrivers() {
     rm -Rf $GPU_DEST /opt/gpu
+    
+    # Remove cached GPU device plugin downloads on non-GPU nodes
+    if [ -d "/opt/nvidia-device-plugin/downloads" ]; then
+        echo "Removing cached ${K8S_DEVICE_PLUGIN_PKG} downloads from non-GPU node..."
+        rm -rf /opt/nvidia-device-plugin/downloads
+        echo "Cached ${K8S_DEVICE_PLUGIN_PKG} downloads removed from non-GPU node"
+    fi
+}
+
+installNvidiaDevicePluginPkgFromCache() {
+    echo "Installing cached nvidia-device-plugin package..."
+    
+    # Find the cached .rpm file with highest version
+    rpmFile=$(find /opt/nvidia-device-plugin/downloads -name "nvidia-device-plugin*.rpm" -type f | sort -V | tail -1)
+    if [ -z "${rpmFile}" ]; then
+        echo "ERROR: No cached nvidia-device-plugin package found" >&2
+        exit $ERR_GPU_DEVICE_PLUGIN_START_FAIL
+    fi
+    
+    echo "Installing ${rpmFile}..."
+    rpm -i "${rpmFile}" || exit $ERR_APT_INSTALL_TIMEOUT
```

Comment:
we  don't have a helper function with retries 

we have installDebPackageFromFile and installRPMPackageFromFile

Please have a look at the PR from Lily, which shows all the existing helper functions